com.unity.render-pipelines.universal/Runtime/UniversalRenderer.cs

```diff
@@ -123,8 +152,29 @@ public UniversalRenderer(UniversalRendererData data) : base(data)
             m_ForwardLights = new ForwardLights();
             //m_DeferredLights.LightCulling = data.lightCulling;
             this.m_RenderingMode = data.renderingMode;
+            this.m_DepthPrepassMode = data.depthPrepassMode;
+            this.m_DepthPrimingMode = data.depthPrimingMode;
             this.usesRenderPass = data.useNativeRenderPass;
 
+#if !UNITY_ANDROID && !UNITY_IOS
+            this.m_CanUseDepthPriming = true;
+
+#if UNITY_EDITOR_WIN || UNITY_STANDALONE_WIN
+            if (CultureInfo.InvariantCulture.CompareInfo.IndexOf(SystemInfo.graphicsDeviceVendor, "ARM", CompareOptions.IgnoreCase) >= 0)
+            {
+                this.m_CanUseDepthPriming = false;
+            }
+#elif UNITY_EDITOR_OSX || UNITY_STANDALONE_OSX
+            if (CultureInfo.InvariantCulture.CompareInfo.IndexOf(SystemInfo.graphicsDeviceVendor, "SILICON", CompareOptions.IgnoreCase) >= 0)
+            {
+                this.m_CanUseDepthPriming = false;
+            }
+#endif
+
+#else
+            this.m_CanUseDepthPriming = false;
+#endif
```

Comment:
I particularly don't like implicit logic like this disabling things under the hood for users. What's the motivation to disable depth priming if we already have a depth prepass?

Reply:
For platforms that don't support https://docs.unity3d.com/ScriptReference/SystemInfo-hasHiddenSurfaceRemovalOnGPU.html it would be improvement to not sort opaques from front-to-back and do more aggresive batching.